### ISSUE TITLE
Cognito Canadian Region

### ISSUE THREAD

**Issue by jblakeney** [spulec/moto#2218]
Due to using an outdated list of supported regions from boto (not boto3), certain valid regions for the Cognito service don't work when mocking up a service :(

Sadly I couldn't find a boto3 compatible alternative to finding valid regions.

```
self = <moto.cognitoidp.responses.CognitoIdpResponse object at 0x7f007ce7a518>

    def create_user_pool(self):
        name = self.parameters.pop("PoolName")
>       user_pool = cognitoidp_backends[self.region].create_user_pool(name, self.parameters)
E       KeyError: 'ca-central-1'
```

**Comment by jblakeney** [comment]
https://github.com/spulec/moto/pull/2226

**Comment by bblommers** [comment]
Support for new regions has been implemented by #2657.

The linked PR also adds support for a new method, so I'll leave that open
